### PR TITLE
ADM-350 Pipeline tool section validation update

### PR DESCRIPTION
**Pull request by spadio9587**
## Summary
update the validation about pipelineTool for token
...

## Before

_Description_

Before

## After

_Description_

After

## Note

_Null_

### CODE REVIEW

File: frontend/src/clients/PipelineToolClient.ts [hunk 1]
```diff
@@ -25,8 +25,8 @@ export class PipelineToolClient extends HttpClient {
       if (code === 400) {
         throw new BadRequestException(params.type, 'Bad request')
       }
-      if (code === 404) {
-        throw new NotFoundException(params.type, 'Page not found')
+      if (code === 401) {
+        throw new NotFoundException(params.type, 'Token is incorrect')
```
Comment (guzhongren) [inline frontend/src/clients/PipelineToolClient.ts:29]: `Incorrect` means the value is not valid
`401` means not `Unauthorized` & the token's format or other things are correct
`NotFoundException` === 401, are u kidding?

**Review by guzhongren — COMMENTED** [review]
Change with the comments

File: frontend/src/clients/PipelineToolClient.ts (hunk 1 above)
Comment (spadio9587) [reply frontend/src/clients/PipelineToolClient.ts:29]: Double check with that, 404 => not found, 401 => Unauthorized, fix it

File: frontend/__tests__/src/components/Metrics/ConfigStep/PipelineTool.test.tsx [hunk 2]
```diff
@@ -110,13 +109,29 @@ describe('PipelineTool', () => {
 
   it('should show error message and error style when token is empty', async () => {
     const { getByText } = setup()
+    const TOKEN_ERROR_MESSAGE = 'Token is required'
+
     await fillPipelineToolFieldsInformation()
 
     const tokenInput = screen.getByTestId('pipelineToolTextField').querySelector('input') as HTMLInputElement
-    await userEvent.type(tokenInput, 'abcd')
+    fireEvent.change(tokenInput, { target: { value: '' } })
```
Comment (Teresasasa) [inline frontend/__tests__/src/components/Metrics/ConfigStep/PipelineTool.test.tsx:117]: replace it with userEvent

**Review by guzhongren — COMMENTED** [review]
change with the comment

File: frontend/src/components/Metrics/ConfigStep/PipelineTool/index.tsx [hunk 3]
```diff
@@ -46,7 +47,17 @@ export const PipelineTool = () => {
...
+    let helperText = ''
+
+    if (value === '') {
+      helperText = TOKEN_HELPER_TEXT.RequiredTokenText
+    } else if (!BUILDKITE_TOKEN_REGEXP.test(value)) {
+      helperText = TOKEN_HELPER_TEXT.InvalidTokenText
+    }
+    setPipelineToolHelperText(helperText)
+    return helperText === ''
+  }
```
Comment (Teresasasa) [inline frontend/src/components/Metrics/ConfigStep/PipelineTool/index.tsx:61]:
```suggestion
    const helperText = !value ? TOKEN_HELPER_TEXT.RequiredTokenText : (!BUILDKITE_TOKEN_REGEXP.test(value) ? TOKEN_HELPER_TEXT.InvalidTokenText : '')
    setPipelineToolHelperText(helperText)
    return !helperText
```

**Review by Teresasasa — APPROVED** [review]
LGTM

**Review by Eason-T — APPROVED** [review]

**Pull request merged**